### DISCUSSION TITLE
Problem solving - generating types from resolver with arguments

### DISCUSSION THREAD

**Discussion by MincePie** [dotansimha/graphql-code-generator#6705] [Q&A] [answered]
I'm trying to follow Ben Awad's lireddit tutorial.

At 6.49 he shows how to add arguments to a resolver query, with types that replace default values (in this case, replacing a default Float value, with Int).

The query is:

```
@Resolver()
  export class PostResolver {
    @Query(() => [Post])
    async posts(
      @Arg("limit", () => Int) limit: number,
      @Arg("cursor", () => String, { nullable: true }) cursor: string | null
    ): Promise<Post[]> {
      const realLimit = Math.min(50, limit);
      
      const qb = getConnection()
        .getRepository(Post)
        .createQueryBuilder("p")
        .orderBy('"createdAt"', "DESC")
        .take(realLimit)
        
        if (cursor) {
          qb.where('"createdAt" < :cursor',  new Date ( parseInt(cursor)) )
        }
       return qb.getMany();
    }
  
```

When I try to generate types from this resolver, I get 3 errors, being:

1. GraphQLDocumentError: Unknown type "Int". [Note: Int is imported from type-graphql]
2. Unknown argument "cursor" on field "Query.posts".
3. Unknown argument "limit" on field "Query.posts".

I'm trying to find things I can do to try and solve this problem.

When I use the playground to inspect the documents, I don't get any arguments feedback. 

This SO post includes a screenshot that shows the feedback I can get from the graphql playground. It doesn't include limit or cursor arguments at all.

**Comment by ardatan** [comment, marked as answer]
I think you should ask this in TypeGraphQL Repo.
https://github.com/MichalLytek/type-graphql
